src/main/java/org/opentripplanner/traffic/SegmentSpeedSample.java

```diff
@@ -0,0 +1,137 @@
+package org.opentripplanner.traffic;
+
+import com.conveyal.traffic.data.ExchangeFormat;
+import com.conveyal.traffic.stats.BaselineStatistics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.time.DayOfWeek;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+/**
+ * Represents speeds at particular times of day.
+ */
+public class SegmentSpeedSample implements Serializable {
+    private static final Logger LOG = LoggerFactory.getLogger(SegmentSpeedSample.class);
+
+    private static final double KMH_TO_MS = 1000d / 3600d;
+
+    /**
+     * the overall average speed on this segment, in centimeters per second, with -32,768 representing 0.
+     * This allows representation of speeds up to 2359 kilometers per hour.
+     */
+    private final short average;
+
+    /**
+     * The average speeds by hour of week, with 0 being midnight Monday morning GMT.
+     * Coded as above.
+     */
+    private final short[] hourBins;
+
+    /** Get a speed estimate in meters per second for the time specified (in milliseconds since the epoch) */
+    public double getSpeed (long time) {
+        if (hourBins == null)
+            return decodeSpeed(average);
+
+        // figure out the hour bin
+        Instant instant = Instant.ofEpochMilli(time);
```

Comment:
This ought to be profiled, I'm a bit wary of the performances here. But AFAIK it can be safely assumed that UNIX epoch modulo 7*24h is the time offset within a week (adjusted for the day of the week of 1/1/70), so simpler arithmetic may be used as a shortcut (see PR #2035).